subprojects/ide-native/src/main/java/org/gradle/ide/visualstudio/internal/DefaultVisualStudioProject.java

```diff
@@ -72,9 +74,8 @@ public void addSourceFile(File sourceFile) {
         additionalFiles.add(sourceFile);
     }
 
-    public String getUuid() {
-        String vsComponentPath = projectPath + ":" + getName();
-        return "{" + UUID.nameUUIDFromBytes(vsComponentPath.getBytes()).toString().toUpperCase() + "}";
+    public static String getUUID(File projectFile) {
```

Comment:
Why did it change to a static method? I see that you use it as a fixture, but for all the other scenario where it has a reference to the project, the call to `getUUID` is more verbose. Maybe having a 2 method, 1) `getUuid()` non-static method and 2) having a static method that (1) delegate to as well as it can be used as a fixture directly.

Reply:
It's used in `VisualStudioSolutionFile` where we need the uuid for the `VisualStudioProject` but we no longer have the relationship (i.e. what we have is a `visualStudioProject` ide artifact, not the original model object.  This is also why we switch to using the projectFile as the key for the uuid rather than the projectPath/name.